File: T-DEV-703/T-DEV-703-Mobile/flutter/lib/models/product.dart
class Product {
  final dynamic id;
  final String name;
  final String barcode;
  final double price;
  final String? image;
  final String shopName;
  final int? quantity;
  final String? nutriscore;
  final String? ingredients;
  final int? novaGroup;
  final dynamic nutrientLevels;

  Product(
      {required this.name,
      required this.barcode,
      required this.price,
      required this.shopName,
      required this.id,
      this.quantity,
      this.nutrientLevels,
      this.novaGroup,
      this.ingredients,
      this.nutriscore,
      this.image});

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
        id: json['id'],
        name: json['name'],
        ingredients: json['ingredients_text_en'] ?? json['ingredients_text'],
        barcode: json['barcode'],
        price: json['priceTTC'],
        quantity: json['stock'],
        image: json['image_url'],
        nutriscore: json['nutriscore_grade'],
        novaGroup: json['nova_group'],
        nutrientLevels: json['nutrient_levels'],
        shopName: json['Shop']['name']);
  }
}

File: T-DEV-703/T-DEV-703-Mobile/front/src/types/product.ts
export interface Product  {
    id?: number;
    barCode?: string;
    name?: string;
    code?: string;
    stock?: number;
    price?: number;
    image?: string;
    image_front_url?: string;
    brands?: string;
    ingredients_text_en?: string;
    nutrient_levels? : {
        fat: string;
        'saturated-fat': string;
        sugars: string;
        salt: string;
    };
    nutriscore_grade?: string;
    nova_group?: number;
    serving_quantity?: string;
    serving_quantity_unit?: string;
    energy_serving?: string;
    energy_unit?: string;
    allergens_from_ingredients?: string;
}